natss/pkg/dispatcher/dispatcher.go

```diff
@@ -300,6 +360,52 @@ func (s *SubscriptionsSupervisor) subscribe(channel eventingchannels.ChannelRefe
 	return &natssSub, nil
 }
 
+func toEvent(msg *stan.Msg) (*cloudevents.Event, error) {
```

Comment:
FYI there is a WIP contribution from the cloudevents community to add the encoding code of stan directly in sdk-go. I think we can work it out on integrating it another PR

Reply:
https://github.com/cloudevents/sdk-go/pull/322 & https://github.com/cloudevents/sdk-go/pull/318

Reply:
Interesting! 
Should we create an issue so that we can keep track of it?

Reply:
you should at least use the Event object and its marshalers. You can drop about 40% of this PR with that code.